src/BCFKeyValueData.cc

```diff
@@ -292,30 +338,40 @@ Status BCFKeyValueData::import_gvcf(const DataCache* cache,
     Status s;
     unique_ptr<BCFWriter> writer;
     unique_ptr<bcf1_t, void(*)(bcf1_t*)> vt(bcf_init(), &bcf_destroy);
-    S(BCFWriter::Open(writer));
 
+    range bucket(-1, 0, GENOMIC_INTERVAL_LEN);
     int c;
-    int chrom_id = -1;
     for(c = bcf_read(vcf.get(), hdr.get(), vt.get());
         c == 0;
         c = bcf_read(vcf.get(), hdr.get(), vt.get())) {
-        // Moved to the next chromosome, write this key.
-        if ( vt->rid != chrom_id &&
-             writer->get_num_entries() > 0) {
-            S(write_chrom(body_->db, writer.get(), dataset, chrom_id));
+        // Make sure a record is not longer than [GENOMIC_INTERVAL_LEN].
+        // If this is not true, then we need to compensate in the search
+        // routine.
+        assert(vt->rlen <= GENOMIC_INTERVAL_LEN);
```

Comment:
Instead of `assert` let's make this case return Status::Invalid with an informative error message.